server/nuxeo-launcher/src/main/java/org/nuxeo/launcher/process/ProcessManager.java

```diff
@@ -1,32 +1,83 @@
-//
-// JODConverter - Java OpenDocument Converter
-// Copyright 2009 Art of Solving Ltd
-// Copyright 2004-2009 Mirko Nasato
-//
-// JODConverter is free software: you can redistribute it and/or
-// modify it under the terms of the GNU Lesser General Public License
-// as published by the Free Software Foundation, either version 3 of
-// the License, or (at your option) any later version.
-//
-// JODConverter is distributed in the hope that it will be useful,
-// but WITHOUT ANY WARRANTY; without even the implied warranty of
-// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-// Lesser General Public License for more details.
-//
-// You should have received a copy of the GNU Lesser General
-// Public License along with JODConverter.  If not, see
-// <http://www.gnu.org/licenses/>.
-//
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Contributors:
+ *      Kevin Leturc <kleturc@nuxeo.com>
+ */
+
 package org.nuxeo.launcher.process;
 
 import java.io.IOException;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang3.SystemUtils;
+
+/**
+ * @since 11.1
+ */
+public class ProcessManager {
+
+    protected final Pattern commandPattern;
 
-public interface ProcessManager {
+    protected ProcessManager(Pattern commandPattern) {
+        this.commandPattern = commandPattern;
+    }
 
-    void kill(Process process, String pid) throws IOException;
+    /**
+     * Finds the pid associated to the regex used to build this manager.
+     */
+    public Optional<String> findPid() throws IOException {
```

Comment:
Why not return an `OptionalLong` instead? It would avoid various map / valueOf stages.

Reply:
No specific reasons, I'll give a try.

Reply:
I'll use `Optional<Long>` due to lack of map/flatMap methods on `OptionalLong`.

In the end, it moves map/valueOf stages from business code to process managers. I'm not sure, we're earning lines there. But in the near future, I would like to remove process managers to leverage only ProcessHandle classes from JDK and it's all about Long, I think we can keep the Long version.

Reply:
Ok it was just a suggestion.